atc/api/jobserver/list_builds.go

```diff
@@ -22,7 +22,7 @@ func (s *Server) ListJobBuilds(pipeline db.Pipeline) http.Handler {
 			limit      int
 		)
 
-		logger := s.logger.Session("list-job-builds")
+		_ = s.logger.Session("list-job-builds")
```

Comment:
I notice you're ignoring this logger, which makes this line a no-op - the returned logger has additional context, which `s.logger` does not.

Rather than adding a `logger` parameter to `logIfErrorAndRespond`, I'd argue that it's better to get rid of `logIfErrorAndRespond` and `checkResultAndRespond`. IMO it's more idiomatic Go to make the error handling obvious and the code simple - even though the current checks are more lines of code, I find it easier to read. It's also easier to modify the behaviour of specific branches (for instance, changing the `logger` that gets used).

That said, if you have a contrary opinion, I'd be very happy to hear you out

Reply:
agree I will address this, I just wanted to display the approach to not duplicate this logger calls, but readability and style of course if affected by this, will address this 👍